src/persist/benches/writer.rs

```diff
@@ -66,40 +130,150 @@ fn block_on<T, F: FnOnce(FutureHandle<T>)>(f: F) -> Result<T, Error> {
     rx.recv()
 }
 
+// Benchmark the write throughput of Indexed::write.
 fn bench_write<L: Log, B: Blob>(
     index: &mut Indexed<L, B>,
     id: Id,
     updates: Vec<((Vec<u8>, Vec<u8>), u64, isize)>,
     b: &mut Bencher,
 ) {
-    b.iter(move || {
-        // We intentionally never call seal, so that the data only gets written
-        // once to Unsealed, and not to Trace.
-        block_on_drain(index, |i, handle| {
-            i.write(vec![(id, updates.clone())], handle)
-        })
-        .unwrap();
+    b.iter_custom(|iters| {
+        // Pre-allocate all of the data we will be writing so that we don't measure
+        // allocation time.
+        let mut data = Vec::with_capacity(iters as usize);
+        for _ in 0..iters {
+            data.push(updates.clone());
+        }
+
+        let start = Instant::now();
+        for _ in 0..iters {
```

Comment:
extra optional nit: alternative way you could write this to avoid the expect is to iterate over data (possibly with an `assert_eq!(data.len(), iters)` right before to make it more obvious, though that's not much different than the assert)

here and elsewhere